tools/docker/Dockerfile.ubuntu-dev

```diff
@@ -0,0 +1,56 @@
+# syntax=docker/dockerfile:1
```

Comment:
unless I misunderstand this what this dockerfile does, I think ubuntu-dev is a wrong name for this file.
Would you like that we publish builder images inside dragonflydb org instead of  bundling them together with the docker release logic?

Reply:
another option is to rename the files to Dockerfile.ubuntu-release (the same applies to alpine of course)

Reply:
To be honest, I'd keep it as-is. That way, the `*-dev` Dockerfiles can progress (if needed), as development goes on. The only additional change, I'd suggest, is building the actual release containers from source as well. That would make it easier, to add support for alpine, for example. 

But I also wanted to have a look at [distroless](https://github.com/GoogleContainerTools/distroless) images, to potentially come up with something even smaller. And with the matrix builds now, throwing in another flavor isn't too difficult :D

The alpine build is already 45% smaller than the ubuntu one :)
```
philipp@ ~ $ docker image ls | grep tamcore/dragonfly
ghcr.io/tamcore/dragonfly   alpha-alpine             36c1d38c6f2e   2 hours ago         107MB
ghcr.io/tamcore/dragonfly   alpha-ubuntu             78dbf284cc5c   2 hours ago         194MB
```

Reply:
FYI, I had trouble with alpine in the past due to its particular choice of using musl library. I do not remember specifics but I refrain from recommending alpine as a default dragonfly container.

Reply:
Regarding your comment - I do not understand it. dev containers are for building dev environment, but you added the next build step to build a container equivalent to a release container, so not i's not a dev container anymore. Right? So what do you mean by "keeping as is" and what do you mean by "building actual release containers from source" ?

Reply:
Fair point. I had a ton of issues trying to get the pre-compiled Dragonfly binaries working under alpine. But so far it seems fine, with the version compiled directly on alpine.

Reply:
> Regarding your comment - I do not understand it. dev containers are for building dev environment, but you added the next build step to build a container equivalent to a release container, so not i's not a dev container anymore. Right? So what do you mean by "keeping as is" and what do you mean by "building actual release containers from source" ?

I guess we're thinking about different things here 😅 

Maybe the name `*-dev` is indeed not optimal. But if we rename them to `*-release`, they kinda obsolete the `*-prod` images, no? Then we could just remove the suffix altogether and have the same Dockerfile for both the weekly and release builds, where the latter would then be built from source as well, rather than relying on the build artifacts from the normal release workflow.

Reply:
hmm, building a binary for aarch64 takes like 2 hours on qemu.... In addition, if we reuse the same binary we will avoid weird synchronization bugs of us having different binaries for the same release. 

Ok, now I understand that dev applies to a release "flavor".  Lets keep the name as is.